Sources/Core/Foundation/HTTPCookieStorageMonitor.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  HTTPCookiesStorageMonitor.swift
+//  XestiMonitors
+//
+//  Created by Angie Mugo on 2018-05-15.
+//
+//  © 2018 J. G. Pusey (see LICENSE.md)
+//
+
+import Foundation
+
+///
+/// A `HTTPCookiesStorageMonitor` instance monitors the cookie storage for changes
+///
+public class HTTPCookiesStorageMonitor: BaseNotificationMonitor {
+    ///
+    /// Encapsulates changes to the cookie storage.
+    ///
+    public enum Event {
+        ///
+        /// The acceptance policy of the cookie storage has changed
+        ///
+        case acceptPolicyChanged(HTTPCookieStorage)
+
+        ///
+        /// The cookies stored in the cookie storage have changed
+        ///
+        case cookiesChanged(HTTPCookieStorage)
+    }
+
+    ///
+    /// Specifies which events to monitor.
+    ///
+    public struct Options: OptionSet {
+        ///
+        /// Monitor `acceptPolicyChanged` events.
+        ///
+        public static let acceptPolicyChanged = Options(rawValue: 1 << 0)
+
+        ///
+        /// Monitor `cookiesChanged` events.
+        ///
+        public static let cookiesChanged = Options(rawValue: 1 << 1)
+
+        ///
+        /// Monitor all events.
+        ///
+        public static let all: Options = [.acceptPolicyChanged,
+                                          .cookiesChanged]
+
+        /// :nodoc:
+        public init(rawValue: UInt) {
+            self.rawValue = rawValue
+        }
+
+        /// :nodoc:
+        public let rawValue: UInt
+    }
+
+    ///
+    /// Initializes a new `HTTPCookiesStorageMonitor`.
+    ///
+    /// - Parameters:
+    ///   - httpCookieStorage:  The cookie storage to monitor.
```

Comment:
Let's call this `cookieStorage` instead (I adjusted the issue wording to match).